src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -37,39 +39,47 @@ class IqPolicyEvaluatorUtil
                                                     final TaskListener listener)
   {
     try {
-      checkArgument(iqPolicyEvaluator.iqStage && iqPolicyEvaluator.iqApplication,
-          'Arguments iqApplication and iqStage are mandatory')
+      String applicationId = iqPolicyEvaluator.getIqApplication()?.applicationId
+
+      checkArgument(iqPolicyEvaluator.iqStage && applicationId, 'Arguments iqApplication and iqStage are mandatory')
+
       LoggerBridge loggerBridge = new LoggerBridge(listener)
       loggerBridge.debug(Messages.IqPolicyEvaluation_Evaluating())
 
-      def iqClient = IqClientFactory.getIqClient(new IqClientFactoryConfiguration(
-          credentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent, log: loggerBridge))
+      def iqClient = IqClientFactory.getIqClient(
+          new IqClientFactoryConfiguration(credentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent,
+              log: loggerBridge))
+
+      def verified = IqUtil.verifyOrCreateApplication(iqClient, applicationId)
+      checkArgument(verified,
+          'Argument iqApplication failed verify or create. To Fix, Either the ' + applicationId +
```

Comment:
I'm not sure we should mention auto application creation here, since all we know for sure is that the application verification failed.  We throw this on the CLI:` log.error("The application ID {} is invalid.", params.getApplicationId());`

Reply:
> I'm not sure we should mention auto application creation here, since all we know for sure is that the application verification failed. We throw this on the CLI:log.error("The application ID {} is invalid.", params.getApplicationId());

I agree. The client doesn't know what happens on the server and it should not know. We should be able to change the server implementation without having to change the client every time.

Reply:
> I'm not sure we should mention auto application creation here, since all we know for sure is that the application verification failed. We throw this on the CLI:log.error("The application ID {} is invalid.", params.getApplicationId());

Changed 7f98e83